### PR TITLE
Fix llvm backend debug info with opts

### PR DESCRIPTION
**Pull request by kripken**
We invoke wasm-opt after finalize, and it needs to be given the source map, and told to emit one, as it changes the binary.

This wasn't noticed in testing because finalize did emit a .map file - it just got out of date later when the binary was changed, but the test just checks for its existence and how many items it has, and doesn't literally understand the source map format and check if things line up to the binary.

cc @yurydelendik

### CODE REVIEW

**Comment by kripken** [comment]
ping @sbc100

File: emcc.py [hunk 1]
```diff
@@ -2672,6 +2675,12 @@ def do_binaryen(target, asm_target, options, memfile, wasm_binary_target,
     cmd += shared.Building.get_binaryen_feature_flags()
     if debug_info:
       cmd += ['-g'] # preserve the debug info
+    if use_source_map(options):
+      cmd += ['--input-source-map=' + wasm_source_map_target]
+      cmd += ['--output-source-map=' + wasm_source_map_target]
```
Comment (sbc100) [inline emcc.py]: We need output-source-map here but not above?    I guess we should factor this out into a "get_binaryen_args()" function at some point?

Comment (kripken) [reply emcc.py]: Where do you mean "above"?

Comment (sbc100) [reply emcc.py]: I mean when we call finalize..around line 2656

Comment (kripken) [reply emcc.py]: the call on 2656 is for asm2wasm, so it's not for the wasm backend. there, there isn't a source map that arrived from the wasm backend that we need to keep passing through.

Comment (sbc100) [reply emcc.py]: OK makes sense.

**Pull request merged**